framework/infrastructure/k8s_internal/k8s_log_collector.py

```diff
@@ -140,3 +151,11 @@ def _write(self, msg: str, force_flush: bool = False):
             self.flush()
         if self.log_to_stdout:
             logger.info(msg)
+
+    def __str__(self):
```

Comment:
Nit: Could add `__repr__ = __str__`

Reply:
Python does something custom for Thread subclasses:
```
In [12]: repr(p)
Out[12]: '<PodLogCollector(pod-log-a, initial daemon)>'

In [13]: str(p)
Out[13]: "PodLogCollector(ident='None', namespace='b', pod_name='a', log_path='c')"
```